hi-ml-cpath/src/health_cpath/configs/classification/BaseMIL.py

```diff
@@ -77,7 +77,8 @@ class BaseMIL(LightningContainer, EncoderParams, PoolingParams):
     def __init__(self, **kwargs: Any) -> None:
         super().__init__(**kwargs)
         self.run_extra_val_epoch = True  # Enable running an additional validation step to save tiles/slides thumbnails
-        self.best_checkpoint_filename = "checkpoint_max_val_auroc"
+        metric_optim = "max" if self.maximise_primary_metric else "min"
+        self.best_checkpoint_filename = f"checkpoint_{metric_optim}_val_{self.primary_val_metric.value}"
         self.best_checkpoint_filename_with_suffix = self.best_checkpoint_filename + ".ckpt"
```

Comment:
shouldn't the equivalent of "checkpoint_max_val_auroc" be `checkpoint_{metric_optim}_val_{self.primary_val_metric.name}`? I guess `self.primary_val_metric.value` will return the value of the metric?

Reply:
`self.primary_val_metric.name` would be  e.g. "ACC" / "ACC_MACRO" whereas `self.primary_val_metric.value` would be "accuracy"/"macro_accuracy" which I think is more readable